app/models/user.rb

```diff
@@ -86,10 +84,23 @@ def password_reset_expired?
   end
 
   def feed page
-    microposts.newest.paginate page: page,
+    following_ids = Relationship.followed_ids(id)
+    Micropost.following_feed(id, following_ids).paginate page: page,
```

Comment:
làm thử theo flow này nhé:
bước 1: lấy ra các user mà thằng self đang follow
bước 2: viết scope để lấy ra các bài viết của tất cả những thằng đó

Reply:
thì dòng này là lấy ra các user mà đang theo dõi mà 
following_ids = Relationship.followed_ids(id)
scope của nó 
scope :followed_ids,
lấy bài viết của người mình theo dõi
    ->(user_id){where(follower_id: user_id).pluck(:followed_id)}
Micropost.following_feed(id, following_ids).paginate page: page,
      per_page: Settings.page_limit.number

scope  
scope :following_feed, (lambda do |user_id, following_ids|
    where(user_id: [user_id, *following_ids])
  end)

Reply:
Micropost.where("user_id IN (
     SELECT followed_id FROM relationships
      WHERE  follower_id = :user_id)
   OR user_id = :user_id", user_id: id)
t tưởng cái này là nó lấy theo user theo id rồi còn gì,  
t ko hiểu rồi ,c giải thích dc ko

Reply:
scope :post_followers, ->(followers){where user_id: followers}
Micropost.post_followers(self.followers).paginate page: page,
t viết lại thế này đúng ko

Reply:
scope :post_followers, ->(following){where user_id: following}
Micropost.post_followers(self.following).paginate page: page,
t viết lại thế này đúng ko